frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts

```diff
@@ -316,6 +322,47 @@ export enum EXPERIMENT_BUTTON_ACTION {
   EXPORT_ALL_EXCLUDE_LISTS = 'export all exclude lists',
 }
 
+export interface UpsertExperimentParams {
+  sourceExperiment: Experiment;
+  action: UPSERT_EXPERIMENT_ACTION;
+}
+
+export interface ExperimentFormData {
+  name: string;
+  description: string;
+  appContext: string;
+  experimentType: ExperimentDesignTypes;
+  unitOfAssignment: ASSIGNMENT_UNIT;
+  consistencyRule: CONSISTENCY_RULE;
+  conditionOrder?: CONDITION_ORDER;
+  assignmentAlgorithm: ASSIGNMENT_ALGORITHM;
+  stratificationFactor?: string;
+  groupType?: string;
+  tags: string[];
+}
+
+export interface AddExperimentRequest {
+  name: string;
+  description: string;
+  context: string[];
+  type: ExperimentDesignTypes;
+  assignmentUnit: ASSIGNMENT_UNIT;
+  consistencyRule: CONSISTENCY_RULE;
+  conditionOrder?: CONDITION_ORDER;
+  assignmentAlgorithm: ASSIGNMENT_ALGORITHM;
+  stratificationFactor?: { stratificationFactorName: string } | null;
+  group?: string;
+  tags: string[];
+  state: EXPERIMENT_STATE;
+  filterMode: FILTER_MODE;
+  // TODO: We might need to add other missing props (see Experiment interface for reference)
+}
+
+// so that we can throw an error if we try to update the id
+export interface UpdateExperimentRequest extends AddExperimentRequest {
+  readonly id: string;
+}
+=======
 export const EXPERIMENT_ROOT_COLUMN_NAMES = {
   NAME: 'name',
```

Comment:
This part (`=======`) got in by mistake when resolving conflicts. I've fixed this in the `experiment-design-refresh` branch.